cmd/agent/app/server.go

```diff
@@ -39,11 +42,19 @@ func (a *Agent) run(o *options.GrpcProxyAgentOptions) error {
 		return fmt.Errorf("failed to validate agent options with %v", err)
 	}
 
-	stopCh := make(chan struct{})
-	if err := a.runProxyConnection(o, stopCh); err != nil {
+	ctx := context.Background()
+	var err error
+	var cs *agent.ClientSet
+	if cs, err = a.runProxyConnection(ctx, o); err != nil {
 		return fmt.Errorf("failed to run proxy connection with %v", err)
 	}
 
+	if features.DefaultMutableFeatureGate.Enabled(features.NodeToMasterTraffic) {
+		if err := a.runControlPlaneProxy(ctx, o, cs); err != nil {
```

Comment:
Might be good to only run the proxy if bind address and mapping are set.

Reply:
Rather than proxy connection, cluster egress might make more sense (as a name)?

Reply:
> Might be good to only run the proxy if bind address and mapping are set.

I'm ok with that if we remove the defaults.

Reply:
done